internal/util/rand.go

```diff
@@ -0,0 +1,48 @@
+//go:build !go1.22
+// +build !go1.22
+
+package util
+
+import (
+	"encoding/binary"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+var sources sync.Pool
+
+func init() {
+	sources = sync.Pool{New: func() any { return rand.New(rand.NewSource(time.Now().UnixNano())) }}
+}
+
+func Shuffle(n int, swap func(i, j int)) {
+	rand.Shuffle(n, swap)
+}
+
+func IntN(n int) int {
+	return rand.Intn(n)
+}
+
+var rngPool = sync.Pool{
+	New: func() any {
+		return rand.New(rand.NewSource(time.Now().UnixNano()))
+	},
+}
+
+func FastRand(n int) (r int) {
+	s := rngPool.Get().(*rand.Rand)
+	r = s.Intn(n)
+	rngPool.Put(s)
+	return
+}
+
+func BinaryString() []byte {
```

Comment:
On self-review, we could probably rename this 
Open to suggestions

Reply:
Indeed. Maybe RandomBytes.